### PR TITLE
feat: add C implementation for `math/base/special/avercos`

### LINKED ISSUE #1714: [RFC]: Add C implementation for `@stdlib/math/base/special/avercos`
### Description

This RFC proposes adding native C implementation for [@stdlib/math/base/special/avercos](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/avercos).

``` c
double stdlib_base_avercos( const double x );
```

### Related Issues

#649

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Snehil-Shah**
Resolves #1714

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for `math/base/special/avercos`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1714
-   progresses #649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/avercos/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2018 The Stdlib Authors.
+Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/avercos/README.md]:
```suggestion
Copyright (c) 2018 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/avercos/test/test.native.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var tape = require( 'tape' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var randu = require( '@stdlib/random/base/randu' );
+var tryRequire = require( '@stdlib/utils/try-require' );
+var EPS = require( '@stdlib/constants/float64/eps' );
+var abs = require( '@stdlib/math/base/special/abs' );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/avercos/test/test.native.js]:
```suggestion
var abs = require( '@stdlib/math/base/special/abs' );
var EPS = require( '@stdlib/constants/float64/eps' );
var tryRequire = require( '@stdlib/utils/try-require' );
```

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Requested minor changes.

**Review by Pranavchiku — APPROVED** [review]
Tested locally, everything works. Thanks for this @Snehil-Shah !

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**